releasenotes/notes/custom-path-returns-e1d5fc1e02809179.yaml

```diff
@@ -0,0 +1,37 @@
+---
+features:
+  - |
+    Two new return types :class:`~retworkx.PathMapping` and
+    :class:`~retworkx.PathLengthMapping`. These classes are returned from
+    functions that previously returened a dictionary of paths or a dictionary
+    of path lengths. They both implement the Python mapping protocol and can
+    be used inplace as a read-only dict.
+upgrade:
+  - |
+    The functions:
+
+    * :func:`retworkx.digraph_k_shortest_path_lengths`
+    * :func:`retworkx.graph_k_shortest_path_lengths`
+    * :func:`retworkx.k_shortest_path_lengths`
+    * :func:`retworkx.digraph_dijkstra_shortest_path_lengths`
+    * :func:`retworkx.graph_dijkstra_shortest_path_lengths`
+    * :func:`retworkx.dijkstra_shortest_path_lengths`
+
+    no longer are returning a ``dict`` and instead are returning
+    :class:`retworkx.PathLengthMapping` objects. This new return type
+    is much faster to build and it implements the python mapping protocol in a
+    read-only fashion and should not be noticeable unless explicit type
+    checking or mutating the result were done.
```

Comment:
It is, this is technically is an API change because the return type changes from a dict to a `PathMapping` or a `PathLengthMapping`, while it should be backwards compatible there are still edge cases that could require manual intervention on upgrading.